src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -288,7 +288,7 @@ private void readAndUpdateMutesRcasDuringStart() {
        */
       if (lastModifiedTimeInMillisInMemory == 0) {
         Set<String> rcasForMute = new HashSet<>(rcaConf.getMutedRcaList());
-        rcasForMute.forEach(mutedRca -> Stats.getInstance().addToMutedGraphNodes(mutedRca));
+        Stats.getInstance().updateMutedGraphNodes(rcasForMute);
```

Comment:
`updateMutedGraphNodes` retains the intersection of the elements in the original and what's provided if provided element is `nonEmpty` or else it adds them all. 

Because the muted rcas in the rca.conf is the final list, we should just clear the old elements and the new ones right ?

Reply:
The `mutedGraphNodes` is initialized as an empty set as part of the RCA Controller `start()`. 

Since `readAndUpdateMutesRcasDuringStart()`  is only invoked during the start, the `mutedGraphNodes` will be an empty set and the  `updateMutedGraphNodes()` on finding `mutedGraphNodes` as empty will add all the elements in rca.conf

Reply:
`Stats` class is static. So, if RCA was disabled, muted rcas changed in the `rca.conf` and later enabled again, then based on this code, `updateMutedGraphNodes` will call `retainAll` that will be lead to an *intersection of the nodes* to be muted.

Reply:
Agreed, this condition was missed. Since rca.conf is our source of truth, now `updateMutedGraphNodes()` is clearing the older values in `mutedGraphNodes` and adding all from rca.conf.